README.md
# Factorio-Tiberium
Note to anyone interested in testing it, it currently doesn't work if it's zipped, for whatever reason.  
Currently trying to figure that one out, -Ax  
  
The tiberium mod for Factorio  
Currently being updated and expanded.

Credits for the foundation we're building on:  
  The Tiberium mod by Zillo7  
  Which was updated and maintained by sah4001 (updated to 0.16) and Oktabyte (updated to 0.17)  


  Command and Conquer Style Defences by Hexicube  
  Who graciously let us adapt and use the 'hardlight walls' code for implementing the Sonic Repulsion Field fences.


changelog
Changed recipe balancing. Uranium is no longer a product of basic ore centrifuging. Uranium tibeirum cost cut in half (5 liquid per ore).

Fixed Tiberium brick to coal using 2 bricks, instead of 1.

Localised various items.

Fixed Node Harvester and Control Network being unlocked at the start *

Made Growth Credit recipes. Added recipes to technologies.


2019.12.05 
	[Art] -Ax
	-Implemented placeholder Tiberium Ore art + glow (Based on Tiberian Sun sprites.)
		* Disabled 'HR rendering' of sprites for now, since there is only the 64px version.
		* 12 variations, 12 growth steps from 1-1000 ore units. For 'smoother' visual spread.
		
		
---------------------------------------------------------------------------------------------------

2019.19.07

Changes Node harvesting recipe, to smooth out production from it. Increased richness of nodes to compensate, since each cycle of the Node harvester harvests one "unit" from the node, regardless of how much it produces.

Added stone growth credit recipe.

Added pollution modifier to molten tib. If ya burn it, the trees will not thank you.

Made growth credit image.

Made icons for each growth credit recipe.

Made icon for Waste recycling recipe.

Changed order of outputs in centrifuging, hopefully to reduce wasted inserter swings. Also removed uranium and oil from output. Direct from slurry only now.

Water recipes commented out. They're largely useless.

Made different icons for the centrifuging recipes.

Added recipe/icon to make stone bricks/concrete out of tiberium waste.

Made new icons for direct recipes.

Fully redid the recipe balancing.

Added a sonic gate.

Removed "Crystal" from localized tib name.

Fully changed liquid tib over to molten.

Added liquid tib fluid.

Fully made the growth accelerator.

Added liquid tib production to make power out of.

Nuclear fuel production from liquid tib, expensive to be efficient.

---------------------------------------------------------------------------------------------------

Jan 8 2020

Fix Node Harvester using 20x power of mining drill for equivalent tib.

Commented out debug info printing to the console.

Removed Tiberium Brick recipes.

Added Tiberium Spikes. They slowly, and forever, produce molten tiberium from nodes, and block growth on them.

Changed Growth Accelerator entity over to centrifuge.

Fixed localization for Sonic Wall Hubs to include distance they work at.

Added Ion Beam turrets, including unique component. The Orbital Ion Cannon uses this unique component instead of laser turrets if you have both mods active.

Moved Sonic walls to using the proper tech, instead of a renamed tech from the orginal mod.

It now takes more time and energy to produce liquid tiberium for your power needs.

All tiberium refining recipes now double the pollution output of the machine.

Redid refining chains, so that refining is aways denser and more valuable.

Waste Processing takes much longer for a single cycle.

---------------------------------------------------------------------------------------------------

Jan 28 2020

Fully updated to 0.18

We now have a Tiberium Damage type. Tiberium inflicts this dmage type now.

Tiberium Armour protects from Tiberium damage.

Tiberium Bullets.

Upped damage from the Ion Projector. Now 4x the vanilla laser turret.

Growth Accelerators now can only be placed on nodes.

Growth Accelerator is immune to tiberium damage.

Tiberium Core missiles. Explosive missiles, with a little bit of liquid tib to give some extra oomph.

Liquid Tiberium bomb is no longer just a nuke. It's a BIG nuke.

Changed Tiberium Ore stack size to 50, to be in line with other ores.

Automated recipe system allows for much more flexible recipes.

---------------------------------------------------------------------------------------------------

Feb 2

Tiered Tiberium centrifuges.

Switched around the products in the centrifuges to promote faster working.

Increased charging speed of Sonic Walls. Reduced required charge. Increased constant drain.

Changed stack size of Growth Credits to 20. Each Growth Accelerator can store 200, which is enough for 2 full growth cycles at max amount in the settings.

Added setting to change how many growth credits can be consumed each cycle. Default 50.

Added setting to change how much damage tiberium does in the various cases. Default 10.

Added setting to add sludge to direct recipes. Default off.

Recipe Updates:

Iron plate upgraded to steel in:Tiberium Node Harvester, Tiberium Spike, Tiberium Control Node.

Electric Mining Drill added to:Tiberium Node Harvester, Tiberium Spike, Tiberium Control Node.

Electric Engine Units added to Tiberium Control Node.

Iron Pipe added to Tiberium Control Node.

Copper Plate added to Sonic Walls.

Advanced Circuits upgraded to processing units in Tiberium Control Node.

Removed Chemical Plant and Electric Engine Unit from Growth Accelerator.


recipe.lua
TibProductivity = {}

table.insert(TibProductivity, "tiberium-ore-processing")
table.insert(TibProductivity, "advanced-tiberium-ore-processing")
table.insert(TibProductivity, "iron-growth-credit")
table.insert(TibProductivity, "copper-growth-credit")
table.insert(TibProductivity, "coal-growth-credit")
table.insert(TibProductivity, "uranium-growth-credit")
table.insert(TibProductivity, "oil-growth-credit")
table.insert(TibProductivity, "energy-growth-credit")

for km, vm in pairs(data.raw.module) do
  if vm.effect.productivity and vm.limitation then
    for _, recipe in ipairs(TibProductivity) do
      table.insert(vm.limitation, recipe)
    end
  end
end

data:extend {
  {
    base_color = {
      b = 0,
      g = 1,
      r = 0
    },
    default_temperature = 75,
    flow_color = {
      b = 0.1,
      g = 1.0,
      r = 0.1
    },
    flow_to_energy_ratio = 0.3,
    heat_capacity = "1KJ",
    icon = "__Factorio-Tiberium__/graphics/icons/fluid/liquid-tiberium.png",
    icon_size = 32,
    max_temperature = 1000,
    name = "liquid-tiberium",
    order = "a[fluid]-c[crude-oil]",
    fuel_value = "50MJ",
	emissions_multiplier = 3,
    pressure_to_speed_ratio = 0.4,
    type = "fluid"
  },
  {
    base_color = {
      b = 0,
      g = 1,
      r = 0
    },
    default_temperature = 75,
    flow_color = {
      b = 0.1,
      g = 1.0,
      r = 0.1
    },
    flow_to_energy_ratio = 0.3,
    heat_capacity = "1KJ",
    icon = "__Factorio-Tiberium__/graphics/icons/fluid/molten-tiberium.png",
    icon_size = 32,
    max_temperature = 1000,
    name = "molten-tiberium",
    order = "a[fluid]-c[crude-oil]",
    fuel_value = "50MJ",
	emissions_multiplier = 3,
    pressure_to_speed_ratio = 0.4,
    type = "fluid"
  },
  {
    base_color = {
      b = 0.3,
      g = 0.5,
      r = 0.3
    },
    default_temperature = 75,
    flow_color = {
      b = 0.3,
      g = 0.5,
      r = 0.3
    },
    flow_to_energy_ratio = 0.2,
    heat_capacity = "1KJ",
    icon = "__Factorio-Tiberium__/graphics/icons/fluid/tiberium-sludge.png",
    icon_size = 32,
    max_temperature = 1000,
    name = "tiberium-sludge",
    order = "a[fluid]-d[crude-oil]",
    pressure_to_speed_ratio = 0.2,
    type = "fluid"
  },
  {
    base_color = {
      b = 0,
      g = 0.7,
      r = 0
    },
    default_temperature = 50,
    flow_color = {
      b = 0,
      g = 0.7,
      r = 0
    },
    flow_to_energy_ratio = 0.5,
    heat_capacity = "1KJ",
    icon = "__Factorio-Tiberium__/graphics/icons/fluid/tiberium-waste.png",
    icon_size = 32,
    max_temperature = 1000,
    name = "tiberium-slurry",
    order = "a[fluid]-d[crude-oil]",
    pressure_to_speed_ratio = 0.05,
    type = "fluid"
  },
  {
    type = "item",
    name = "tiberium-brick",
    icon = "__Factorio-Tiberium__/graphics/icons/tiberium-brick.png",
    icon_size = 32,
    flags = {},
    subgroup = "intermediate-product",
    stack_size = 100
  }
}

data:extend(
  {
    {
      type = "recipe",
      name = "tiberium-ore-processing",
      category = "oil-processing",
      energy_required = 5,
      enabled = false,
      ingredients = {
        {type = "fluid", name = "water", amount = 100},
        {type = "item", name = "tiberium-ore", amount = 5}
      },
      results = {
        {type = "fluid", name = "tiberium-slurry", amount = 3},
      },
      icon = "__Factorio-Tiberium__/graphics/icons/fluid/tiberium-waste.png",
      icon_size = 32,
      subgroup = "a-refining",
      order = "b[fluid-chemistry]-f[heavy-oil-cracking]"
    },
    {
      type = "recipe",
      name = "advanced-tiberium-ore-processing",
      category = "oil-processing",
      energy_required = 5,
      enabled = false,
      ingredients = {
        {type = "fluid", name = "water", amount = 50},
        {type = "fluid", name = "tiberium-slurry", amount = 3},
      },
      results = {
        {type = "fluid", name = "molten-tiberium", amount = 4}
      },
      icon = "__Factorio-Tiberium__/graphics/icons/fluid/molten-tiberium.png",
      icon_size = 32,
      subgroup = "a-refining",
      order = "b[fluid-chemistry]-f[heavy-oil-cracking]"
    },
	
    {
      type = "recipe",
      name = "advanced-tiberium-brick-processing",
      category = "crafting-with-fluid",
      energy_required = 2,
      enabled = false,
      ingredients = {
        {type = "fluid", name = "molten-tiberium", amount = 3}
      },
      results = {
        {type = "item", name = "tiberium-brick", amount = 2}
      },
      icon = "__Factorio-Tiberium__/graphics/icons/tiberium-brick.png",
      icon_size = 32,
      subgroup = "a-refining",
      order = "b[oil-processing]-h[advanced-oil-processing]"
    },
	{
      type = "recipe",
      name = "tiberium-liquid-processing",
      category = "chemistry",
      energy_required = 20,
      enabled = false,
      ingredients = {
        {type = "fluid", name = "molten-tiberium", amount = 3},
      },
      results = {
        {type = "fluid", name = "liquid-tiberium", amount = 1}
      },
      icon = "__Factorio-Tiberium__/graphics/icons/fluid/liquid-tiberium.png",
      icon_size = 32,
      subgroup = "a-refining",
      order = "m"
    },
    {
      type = "recipe",
      name = "tiberium-brick-to-iron-ore",
      category = "chemistry",
      energy_required = 4,
      enabled = false,
      ingredients = {
        {type = "item", name = "tiberium-brick", amount = 1}
      },
      results = {
        {type = "item", name = "iron-ore", amount = 4}
      },
      icon = "__Factorio-Tiberium__/graphics/icons/tiberium-brick-to-iron.png",
      icon_size = 32,
      subgroup = "a-direct",
      order = "b[oil-processing]-e[advanced-oil-processing]"
    },
    {
      type = "recipe",
      name = "tiberium-brick-to-copper-ore",
      category = "chemistry",
      energy_required = 4,
      enabled = false,
      ingredients = {
        {type = "item", name = "tiberium-brick", amount = 1}
      },
      results = {
        {type = "item", name = "copper-ore", amount = 4}
      },
      icon = "__Factorio-Tiberium__/graphics/icons/tiberium-brick-to-copper.png",
      icon_size = 32,
      subgroup = "a-direct",
      order = "b[oil-processing]-f[advanced-oil-processing]"
    },
    {
      type = "recipe",
      name = "tiberium-brick-to-coal",
      category = "chemistry",
      energy_required = 4,
      enabled = false,
      ingredients = {
        {type = "item", name = "tiberium-brick", amount = 1}
      },
      results = {
        {type = "item", name = "coal", amount = 4}
      },
      icon = "__Factorio-Tiberium__/graphics/icons/tiberium-brick-to-coal.png",
      icon_size = 32,
      subgroup = "a-direct",
      order = "b[oil-processing]-g[advanced-oil-processing]"
    },
    {
      type = "recipe",
      name = "tiberium-brick-to-stone",
      category = "chemistry",
      energy_required = 4,
      enabled = false,
      ingredients = {
        {type = "item", name = "tiberium-brick", amount = 1}
      },
      results = {
        {type = "item", name = "stone", amount = 4}
      },
      icon = "__Factorio-Tiberium__/graphics/icons/tiberium-brick-to-stone.png",
      icon_size = 32,
      subgroup = "a-direct",
      order = "b[oil-processing]-d[petroleum-gas-processing]"
    },
	{
      type = "recipe",
      name = "tiberium-sludge-to-stone-brick",
      category = "chemistry",
      energy_required = 4,
      enabled = false,
      ingredients = {
        {type = "fluid", name = "tiberium-sludge", amount = 1}
      },
      results = {
        {type = "item", name = "stone-brick", amount = 1}
      },
      icon = "__Factorio-Tiberium__/graphics/icons/tiberium-sludge-to-stone-brick.png",
      icon_size = 32,
      subgroup = "a-direct",
      order = "z"
    },
    --[[{
      type = "recipe",
      name = "tiberium-brick-to-water",
      category = "chemistry",
      energy_required = 0.1,
      enabled = false,
      ingredients = {
        {type = "item", name = "tiberium-brick", amount = 1}
      },
      results = {
        {type = "fluid", name = "water", amount = 1000}
      },
      icon = "__base__/graphics/icons/fluid/water.png",
      icon_size = 32,
      subgroup = "a-direct",
      order = "b[oil-processing]-g[light-oil-processing]"
    },]]
    {
      type = "recipe",
      name = "tiberium-molten-to-iron-ore",
      category = "chemistry",
      energy_required = 4,
      enabled = false,
      ingredients = {
        {type = "fluid", name = "molten-tiberium", amount = 3}
      },
      results = {
        {type = "item", name = "iron-ore", amount = 4}
      },
      icon = "__Factorio-Tiberium__/graphics/icons/molten-to-iron.png",
      icon_size = 32,
      subgroup = "a-direct",
      order = "b[oil-processing]-e[advanced-oil-processing]"
    },
    {
      type = "recipe",
      name = "tiberium-molten-to-copper-ore",
      category = "chemistry",
      energy_required = 4,
      enabled = false,
      ingredients = {
        {type = "fluid", name = "molten-tiberium", amount = 3}
      },
      results = {
        {type = "item", name = "copper-ore", amount = 4}
      },
      icon = "__Factorio-Tiberium__/graphics/icons/molten-to-copper.png",
      icon_size = 32,
      subgroup = "a-direct",
      order = "b[oil-processing]-f[advanced-oil-processing]"
    },
    {
      type = "recipe",
      name = "tiberium-molten-to-coal",
      category = "chemistry",
      energy_required = 4,
      enabled = false,
      ingredients = {
        {type = "fluid", name = "molten-tiberium", amount = 3}
      },
      results = {
        {type = "item", name = "coal", amount = 4}
      },
      icon = "__Factorio-Tiberium__/graphics/icons/molten-to-coal.png",
      icon_size = 32,
      subgroup = "a-direct",
      order = "b[oil-processing]-g[advanced-oil-processing]"
    },
    {
      type = "recipe",
      name = "tiberium-molten-to-stone",
      category = "chemistry",
      energy_required = 4,
      enabled = false,
      ingredients = {
        {type = "fluid", name = "molten-tiberium", amount = 3}
      },
      results = {
        {type = "item", name = "stone", amount = 4}
      },
      icon = "__Factorio-Tiberium__/graphics/icons/molten-to-stone.png",
      icon_size = 32,
      subgroup = "a-direct",
      order = "b[oil-processing]-d[petroleum-gas-processing]"
    },
    {
      type = "recipe",
      name = "tiberium-slurry-to-uranium-ore",
      category = "chemistry",
      energy_required = 5,
      enabled = false,
      ingredients = {
        {type = "fluid", name = "tiberium-slurry", amount = 15}
      },
      results = {
        {type = "item", name = "uranium-ore", amount = 6}
      },
      icon = "__Factorio-Tiberium__/graphics/icons/slurry-to-uranium.png",
      icon_size = 32,
      subgroup = "a-direct",
      order = "b[oil-processing]-y[petroleum-gas-processing]"
    },
    {
      type = "recipe",
      name = "tiberium-slurry-to-crude-oil",
      category = "chemistry",
      energy_required = 4,
      enabled = false,
      ingredients = {
        {type = "fluid", name = "tiberium-slurry", amount = 3}
      },
      results = {
        {type = "fluid", name = "crude-oil", amount = 40}
      },
      icon = "__Factorio-Tiberium__/graphics/icons/slurry-to-oil.png",
      icon_size = 32,
      subgroup = "a-direct",
      order = "b[oil-processing]-z[heavy-oil-processing]"
    },
    --[[{
      type = "recipe",
      name = "tiberium-molten-to-water",
      category = "chemistry",
      energy_required = 0.1,
      enabled = false,
      ingredients = {
        {type = "fluid", name = "molten-tiberium", amount = 1}
      },
      results = {
        {type = "fluid", name = "water", amount = 1000}
      },
      icon = "__base__/graphics/icons/fluid/water.png",
      icon_size = 32,
      subgroup = "a-direct",
      order = "b[oil-processing]-g[light-oil-processing]"
    },]]
    {
      type = "recipe",
      name = "tiberium-ore-centrifuging",
      category = "centrifuging",
	  subgroup = "a-centrifuging",
      energy_required = 5,
      enabled = false,
      ingredients = {
        {type = "fluid", name = "water", amount = 100},
        {type = "item", name = "tiberium-ore", amount = 34}
      },
      results = {
		{type = "item", name = "stone", amount = 1},
		{type = "item", name = "coal", amount = 1},
        {type = "item", name = "copper-ore", amount = 6},
		{type = "item", name = "iron-ore", amount = 8}
      },
      icon = "__Factorio-Tiberium__/graphics/icons/tiberium-centrifuging.png",
      icon_size = 32,
      order = "a[fluid-chemistry]-f[heavy-oil-cracking]"
    },
    {
      type = "recipe",
      name = "tiberium-slurry-centrifuging",
      category = "centrifuging",
	  subgroup = "a-centrifuging",
      energy_required = 4,
      enabled = false,
      ingredients = {
        {type = "fluid", name = "tiberium-slurry", amount = 22}
      },
      results = {
		{type = "item", name = "stone", amount = 1},
		{type = "item", name = "coal", amount = 1},
        {type = "item", name = "copper-ore", amount = 6},
		{type = "item", name = "iron-ore", amount = 8}
      },
      icon = "__Factorio-Tiberium__/graphics/icons/slurry-centrifuging.png",
      icon_size = 32,
      order = "b[fluid-chemistry]-f[heavy-oil-cracking]"
    },
    {
      type = "recipe",
      name = "tiberium-molten-centrifuging",
      category = "centrifuging",
	  subgroup = "a-centrifuging",
      energy_required = 4,
      enabled = false,
      ingredients = {
        {type = "fluid", name = "molten-tiberium", amount = 22}
      },
      results = {
		{type = "item", name = "stone", amount = 1},
		{type = "item", name = "coal", amount = 1},
        {type = "item", name = "copper-ore", amount = 6},
		{type = "item", name = "iron-ore", amount = 8}
      },
      icon = "__Factorio-Tiberium__/graphics/icons/molten-centrifuging.png",
      icon_size = 32,
      order = "c[fluid-chemistry]-f[heavy-oil-cracking]"
    },
	{
      type = "recipe",
      name = "tiberium-ore-sludge-centrifuging",
      category = "centrifuging",
	  subgroup = "a-centrifuging",
      energy_required = 5,
      enabled = false,
      ingredients = {
        {type = "fluid", name = "water", amount = 100},
        {type = "item", name = "tiberium-ore", amount = 34}
      },
      results = {
		{type = "fluid", name = "tiberium-sludge", amount = 1},
		{type = "item", name = "coal", amount = 1},
        {type = "item", name = "copper-ore", amount = 6},
		{type = "item", name = "iron-ore", amount = 8}
      },
      icon = "__Factorio-Tiberium__/graphics/icons/tiberium-centrifuging.png",
      icon_size = 32,
      order = "d"
    },
    {
      type = "recipe",
      name = "tiberium-slurry-sludge-centrifuging",
      category = "centrifuging",
	  subgroup = "a-centrifuging",
      energy_required = 4,
      enabled = false,
      ingredients = {
        {type = "fluid", name = "tiberium-slurry", amount = 22}
      },
      results = {
		{type = "fluid", name = "tiberium-sludge", amount = 1},
		{type = "item", name = "coal", amount = 1},
        {type = "item", name = "copper-ore", amount = 6},
		{type = "item", name = "iron-ore", amount = 8}
      },
      icon = "__Factorio-Tiberium__/graphics/icons/slurry-centrifuging.png",
      icon_size = 32,
      order = "e"
    },
    {
      type = "recipe",
      name = "tiberium-molten-sludge-centrifuging",
      category = "centrifuging",
	  subgroup = "a-centrifuging",
      energy_required = 4,
      enabled = false,
      ingredients = {
        {type = "fluid", name = "molten-tiberium", amount = 22}
      },
      results = {
		{type = "fluid", name = "tiberium-sludge", amount = 1},
		{type = "item", name = "coal", amount = 1},
        {type = "item", name = "copper-ore", amount = 6},
		{type = "item", name = "iron-ore", amount = 8}
      },
      icon = "__Factorio-Tiberium__/graphics/icons/molten-centrifuging.png",
      icon_size = 32,
      order = "f"
    },
    {
      type = "recipe",
      name = "tiberium-waste-recycling",
      category = "chemistry",
      energy_required = 2,
      enabled = false,
      ingredients = {
        {type = "fluid", name = "molten-tiberium", amount = 20},
        {type = "fluid", name = "tiberium-sludge", amount = 5}
      },
      results = {
        {type = "fluid", name = "molten-tiberium", amount = 23}
      },
      icon = "__Factorio-Tiberium__/graphics/icons/tiberium-recycling.png",
      icon_size = 32,
      subgroup = "a-refining",
      order = "b[fluid-chemistry]-f[heavy-oil-cracking]"
    },
	{
      type = "recipe",
      name = "iron-growth-credit",
      category = "chemistry",
	  subgroup = "a-growth-credits",
      energy_required = 20,
      enabled = false,
      ingredients = {
        {type = "item", name = "iron-ore", amount = 106}
      },
      results = {
        {type = "item", name = "growth-credit", amount = 1},
      },
      icon = "__Factorio-Tiberium__/graphics/icons/iron-growth-credit.png",
      icon_size = 32,
      order = "b[fluid-chemistry]-f[heavy-oil-cracking]"
    },
	{
      type = "recipe",
      name = "copper-growth-credit",
      category = "chemistry",
	  subgroup = "a-growth-credits",
      energy_required = 20,
      enabled = false,
      ingredients = {
        {type = "item", name = "copper-ore", amount = 106}
      },
      results = {
        {type = "item", name = "growth-credit", amount = 1},
      },
      icon = "__Factorio-Tiberium__/graphics/icons/copper-growth-credit.png",
      icon_size = 32,
      order = "b[fluid-chemistry]-f[heavy-oil-cracking]"
    },
	{
      type = "recipe",
      name = "coal-growth-credit",
      category = "chemistry",
	  subgroup = "a-growth-credits",
      energy_required = 20,
      enabled = false,
      ingredients = {
        {type = "item", name = "coal", amount = 106}
      },
      results = {
        {type = "item", name = "growth-credit", amount = 1},
      },
      icon = "__Factorio-Tiberium__/graphics/icons/coal-growth-credit.png",
      icon_size = 32,
      order = "b[fluid-chemistry]-f[heavy-oil-cracking]"
    },
	{
      type = "recipe",
      name = "uranium-growth-credit",
      category = "chemistry",
	  subgroup = "a-growth-credits",
      energy_required = 20,
      enabled = false,
      ingredients = {
        {type = "item", name = "uranium-ore", amount = 32}
      },
      results = {
        {type = "item", name = "growth-credit", amount = 1},
      },
      icon = "__Factorio-Tiberium__/graphics/icons/uranium-growth-credit.png",
      icon_size = 32,
      order = "b[fluid-chemistry]-f[heavy-oil-cracking]"
    },
	{
      type = "recipe",
      name = "stone-growth-credit",
      category = "chemistry",
	  subgroup = "a-growth-credits",
      energy_required = 20,
      enabled = false,
      ingredients = {
        {type = "item", name = "stone", amount = 106}
      },
      results = {
        {type = "item", name = "growth-credit", amount = 1},
      },
      icon = "__Factorio-Tiberium__/graphics/icons/stone-growth-credit.png",
      icon_size = 32,
      order = "b[fluid-chemistry]-f[heavy-oil-cracking]"
    },
	{
      type = "recipe",
      name = "oil-growth-credit",
      category = "chemistry",
	  subgroup = "a-growth-credits",
      energy_required = 20,
      enabled = false,
      ingredients = {
        {type = "fluid", name = "crude-oil", amount = 1060}
      },
      results = {
        {type = "item", name = "growth-credit", amount = 1},
      },
      icon = "__Factorio-Tiberium__/graphics/icons/oil-growth-credit.png",
      icon_size = 32,
      order = "b[fluid-chemistry]-f[heavy-oil-cracking]"
    },
	{
      type = "recipe",
      name = "energy-growth-credit",
      category = "chemistry",
	  subgroup = "a-growth-credits",
      energy_required = 200,
      enabled = false,
      ingredients = {
      },
      results = {
        {type = "item", name = "growth-credit", amount = 1},
      },
      icon = "__Factorio-Tiberium__/graphics/icons/energy-growth-credit.png",
      icon_size = 32,
      order = "b[fluid-chemistry]-f[heavy-oil-cracking]"
    },
	{
      type = "recipe",
      name = "liquid-tiberium-cell",
      category = "chemistry",
	  subgroup = "a-items",
      energy_required = 20,
      enabled = false,
      ingredients = {
		{type = "fluid", name = "liquid-tiberium", amount = 102},
		{type = "item", name = "steel-plate", amount = 2},
		{type = "item", name = "copper-plate", amount = 2},
        {type = "item", name = "electric-engine-unit", amount = 1},
        {type = "item", name = "processing-unit", amount = 2},
        {type = "item", name = "pipe", amount = 1},
      },
      results = {
        {type = "item", name = "nuclear-fuel", amount = 1},
      },
      icon = "__base__/graphics/icons/nuclear-fuel.png",
      icon_size = 32,
      order = "b[fluid-chemistry]-f[heavy-oil-cracking]"
    },
	{
		type = "recipe",
		name = "CnC_SonicWall_Hub",
        enabled = false,
		energy_required = 5,
		ingredients = {{"steel-plate", 25}, {"advanced-circuit", 10}, {"battery", 10}},
		result = "CnC_SonicWall_Hub"
	},
  }
)
